main/ejml-ddense/src/org/ejml/dense/row/mult/MatrixVectorMult_DDRM.java

```diff
@@ -346,4 +348,85 @@ public static double innerProduct( double a[], int offsetA,
 
         return output;
     }
+
+    /**
+     * scalar = A<sup>T</sup>*B*C
+     *
+     * @param A (Input) vector
+     * @param B (Input) Matrix
+     * @param C (Input) vector
+     */
+    public static double innerProduct(DMatrixSparse A, DMatrix1Row B, DMatrixSparse C) {
+        if (A.getNumElements() < B.numRows)
+            throw new IllegalArgumentException("Length of 'a' isn't long enough");
+        if (C.getNumElements() < B.numCols)
+            throw new IllegalArgumentException("Length of 'c' isn't long enough");
+
+        VectorEntry[] nzValuesC = nzValues(C);
+        int sizeC = nzValuesC.length;
+
+        double output = 0.0;
+        for (Iterator<DMatrixSparse.CoordinateRealValue> i = A.createCoordinateIterator(); i.hasNext(); ) {
```

Comment:
Could you add another version that is specific to `DMatrixSparseCSC`  that directly accesses the non-zero values? I bet that will be a lot faster. It would make sense to keep both versions around.

Reply:
Excellent suggestion.  My benchmarking shows ~10% increased throughput with the direct access to the non-zero values.

Reply:
Was expecting it to be more, but still an improvement!